### ISSUE TITLE
Rsyslog not writing to the file when highwatermark reached

### ISSUE THREAD

**Issue by tomassykora** [rsyslog/rsyslog#1337]
Having this configuration:

$ModLoad imuxsock
$ModLoad imklog
$WorkDirectory /var/spool/rsyslog
$ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat

$ActionQueueType LinkedList
$ActionQueueHighWaterMark 700
$ActionQueueFileName remoteq
$ActionQueueSize 10000
$ActionQueueMaxFileSize 1g
$ActionResumeRetryCount -1
$ActionResumeInterval 20

local2.error    @@$SOME_UNREACHABLE_ADDRESS

rsyslog should write to a file when highwatermark is reached. Instead of that, rsyslog writes to the file when it is stopped. Is this a bug or just a change in behaviour?

**Comment by rgerhards** [comment]
Is there any other input involved, namely TCP?

Sent from phone, thus brief.

Am 19.12.2016 12:50 schrieb "tomassykora" <notifications@github.com>:

**Comment by tomassykora** [comment]
No, this is the whole rsyslog.conf. I'm just sending messages (5000) with logger. I suppose they should be immediately written to the queue file, but this doesn't happen until rsyslog is stopped.

**Comment by rgerhards** [comment]
So logger does block and wait?

Sent from phone, thus brief.

Am 19.12.2016 14:47 schrieb "tomassykora" <notifications@github.com>:

**Comment by rgerhards** [comment]
And just to be sure: there is no systemd journal involved at all on the
system in question? I ask because I do not see imjournal.

Sent from phone, thus brief.

Am 19.12.2016 14:55 schrieb "Rainer Gerhards" <rgerhards@hq.adiscon.com>:

**Comment by tomassykora** [comment]
No, logger doesn't block, it sends all messages. But they won't appear in the queue file until rsyslog is stopped. Systemd is in the system but in this specific failing testcase imjournal is not used, just the two modules from the conf I posted.

**Comment by tomassykora** [comment]
I maybe found the problem. In function actionResetQueueParams(viod) highWtrMark and lowWtrMark are set to -1. When I set them to some value here (800/200), it starts to work fine.

**Comment by rgerhards** [comment]
That's interesting. In theory, IIRC, when set to -1 the settings should be auto-adjusted.
